tests/system/fim/common_tasks/verify_alerts_elasticsearch.py

```diff
@@ -438,27 +480,43 @@ def verify_es_alerts(files_list, max_retry, query, no_alert_style, es, index_nam
         # select the scenario
         scenario = select_scenario(scenario_arg_dic)
         scenario_arg = scenario_arg_dic[scenario]
-        print("The selected scenario is {}, and scenario arg is {}".format(scenario,scenario_arg))
+        logging.info("The selected scenario is {}, and scenario arg is {}".format(scenario,scenario_arg))
 
     # read the list of paths from a file into a list
     files_list = read_file(args.files)
-
+    expected_alerts_num = len(files_list)
     # alerts verification
     success, failure, failure_list = \
         verify_es_alerts(files_list, args.max_retry, query, args.no_alert_style,
                          es, index_name, start, args.sleep_time,scenario, scenario_arg)
 
-    elapsed = start - time()
+    elapsed = (datetime.datetime.now().replace(microsecond=0)) - start
+
     with open(args.output, 'w+') as output:
         output.writelines('\n'.join(failure_list))
 
-    assert failure == 0, "number of failed files: {}\n \
-            Elapsed time: ~ {} seconds.".format(
-            failure, elapsed
-        )
+    passed = len(failure_list) == 0    
+    received_alerts_num = expected_alerts_num - len(failure_list)
 
-    print(
-        "Number of succeded files: {}\n Elapsed time: ~ {} seconds.".format(
-            success, elapsed
+    logging.info(
+        "Number of succeeded files: {}/{}. Elapsed time: {}".format(
+            success, expected_alerts_num, elapsed
         )
-    )
\ No newline at end of file
+    )
+
+    try:
+        assert failure == 0
+    except:
+        logging.error("Verification result is FAILED. Number of failed paths: {}/{}"\
+            .format(failure, expected_alerts_num))
+    finally:
```

Comment:
Regular program logic should not be inside the `finally` clause. Use this only for closing/releasing resources.

Reply:
Hi @xr09 !

I think including the results files generation at `finally` is a secure way to ensure that the result file will always be generated however the script fails at some point.

Thank you,

Rshad